### PR TITLE
Some small fixes around .reduce

### PR DESCRIPTION
**Pull request by fregante**
Tiny changes only

### CODE REVIEW

File: src/nativeEditor/Overlay.tsx [hunk 1]
```diff
@@ -372,21 +372,23 @@ function getBoundingClientRectWithBorderOffset(node: HTMLElement) {
...
-    };
-  });
+  for (const rect of rects) {
+    merged.top += rect.top;
+    merged.left += rect.left;
+    merged.bottom += rect.bottom;
+    merged.right += rect.right;
+  }
+
+  return merged;
```
[the remark this answers is not in the document]
Comment (twschiller) [reply src/nativeEditor/Overlay.tsx:391]: This code is copied from an MIT-licensed Facebook repo, so generally don't want to make many changes to it to make it easier pull in upstream changes later https://github.com/facebook/react/blob/b438699d3620bff236282b049204e1221b3689e9/packages/react-devtools-shared/src/backend/views/Highlighter/Overlay.js

File: src/nativeEditor/infer.ts [hunk 2]
```diff
@@ -101,11 +94,8 @@ function commonAttr($items: JQuery<HTMLElement>, attr: string) {
 
   // For classes and rel we take the common values
   if (MULTI_ATTRS.includes(attr)) {
-    const classNames = attributeValues.map(
-      (x) => new Set(x ? x.split(" ") : [])
-    );
-    const commonValues = intersection(classNames);
-    unfiltered = [...commonValues.values()];
+    const classNames = attributeValues.map((x) => (x ? x.split(" ") : []));
+    unfiltered = uniq(intersection(...classNames));
```
Comment (twschiller) [inline src/nativeEditor/infer.ts:98]: I don't think the uniq is necessary: https://lodash.com/docs/4.17.15#intersection

File: src/blocks/readers/ArrayCompositeReader.ts [hunk 3]
```diff
@@ -40,17 +40,15 @@ class ArrayCompositeReader extends Reader {
...
       $schema: "https://json-schema.org/draft/2019-09/schema#",
       type: "object",
-      // eslint-disable-next-line unicorn/no-array-reduce
-      properties: this._readers.reduce(
-        (acc, reader) => ({
-          ...acc,
-          ...(reader.outputSchema.properties ?? {}),
-        }),
-        {}
-      ),
```
[the remark this answers is not in the document]
Comment (twschiller) [reply src/blocks/readers/ArrayCompositeReader.ts:53]: In general, let's prioritize readability over performance unless we observe the performance being a problem

**Review by twschiller — APPROVED** [review]

File: src/nativeEditor/Overlay.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (fregante) [reply src/nativeEditor/Overlay.tsx:391]: Should we move this into the vendor folder in that case?

File: src/blocks/readers/ArrayCompositeReader.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (fregante) [reply src/blocks/readers/ArrayCompositeReader.ts:53]: Yes definitely, readability *and* performance are the reason why I don’t use reduce.

**Pull request merged**